pyccl/halos/profiles.py

```diff
@@ -147,6 +147,38 @@ def _get_plaw_projected(self, cosmo, a):
         """
         return self.precision_fftlog['plaw_projected']
 
+    def __eq__(self, prof2):
+        """ Return `True` if this profile is equivalent to another."""
+        if type(self) != type(prof2):
+            # catch different profile types
+            return False
+
+        params, params2 = self.__dict__, prof2.__dict__
+
+        if params == params2:
+            # same profiles
+            return True
+
+        for key, val in params.items():
+            if key == "cM":
+                cm2 = params2.get(key)
+                if val.name != cm2.name:
+                    return False
+                if not val.mdef.__eq__(cm2.mdef):
+                    return False
+            elif key == "precision_fftlog":
+                if val != params2[key]:
+                    return False
+            else:
+                if val != params2.get(key):
+                    return False
```

Comment:
Yes, good catch, I didn't know dictionary equality can check more than one level in depth. I removed this clause.

Reply:
Yes indeed, good catch, I removed it.

Reply:
Yes, the first check is for speed. In case the `__dict__` of the profiles only contain numerical values and no objects (such as `cM`), this check speeds up the entire process.

Reply:
OK, I see.